arch/risc-v/src/mpfs/mpfs_fpga_canfd.c

```diff
@@ -1150,126 +1181,89 @@ static int mpfs_interrupt(int irq, void *context, void *arg)
 
   uint32_t isr;
   uint32_t icr;
-  uint32_t imask;
-  int irq_loops;
-
-  for (irq_loops = 0; irq_loops < 10000; irq_loops++)
-    {
-      /* Get the interrupt status */
-
-      isr  = getreg32(priv->base + MPFS_CANFD_INT_STAT_OFFSET);
-
-      /* Check and exit interrupt service routine if there is no int flag in
-       * INT_STAT reg.
-       */
-
-      if (!isr)
-        {
-          return irq_loops ? OK : -1;
-        }
 
-      /* Receive Buffer Not Empty Interrupt */
+  /* Get the interrupt status */
 
-      if (isr & MPFS_CANFD_INT_STAT_RBNEI)
-        {
-          /* Mask RBNEI first, then clear interrupt. Even if
-          * another IRQ fires, RBNEI will always be 0 (masked).
-          */
+  isr  = getreg32(priv->base + MPFS_CANFD_INT_STAT_OFFSET);
+  priv->can.isr_status = isr;
 
-          icr = MPFS_CANFD_INT_STAT_RBNEI;
-          putreg32(icr, priv->base + MPFS_CANFD_INT_MASK_SET_OFFSET);
-          putreg32(icr, priv->base + MPFS_CANFD_INT_STAT_OFFSET);
+  /* Receive overrun interrupt. This is handled in mpfs_receive_work.
+   * In case of DOI, also RB has data so RBNEI is set, so work is
+   * always queued
+   */
 
-          /* Schedule work to process RX frame from CAN controller */
+  if (isr & MPFS_CANFD_INT_STAT_DOI)
```

Comment:
This could be done in the errors function, I'll move it in there to keep it consistent